model_analyzer/config/run/run_config_generator.py

```diff
@@ -36,21 +41,114 @@ def __init__(self, config, client):
             TritonClient to be used for interacting with the Triton API
         """
 
+        self._config = config
         self._analyzer_config = config.get_all_config()
+        self._run_search = RunSearch(config=config)
         self._run_configs = []
         self._client = client
         self._model_name_index = 0
         self._model_configs = []
 
-    def run_configs(self):
+    def init(self, model):
+
+        if self._config.run_config_search_disable:
+            logger.info(
+                f"Running manual config search for model: {model.model_name()}")
+            self._run_model_no_search(model)
+        else:
+            logger.info(
+                f"Running auto config search for model: {model.model_name()}")
+            self._run_model_with_search(model)
+
+    def is_done(self):
+        """ Return true if all RunConfigs for the model have been returned """
+        return len(self._run_configs) == 0
+
+    def add_measurement(self, measurement):
+        """ Gives the results from the last RunConfig """
```

Comment:
This comment is incorrect now. This method isnt returning anything.

Reply:
The comment isn't incorrect, I just think I poorly worded it.  It is given the results of the last RunConfig.  I'll clean it up. Thanks for reviewing!